### DISCUSSION TITLE
How to externalitze domain events to Kafka / AMQP, etc.?

### DISCUSSION THREAD

**Discussion by matiwinnetou** [spring-projects/spring-modulith#448] [Q&A] [answered]
Spring Modulith by default supports internal events to be persisted via JPA / JDBC into SQLalike supported DBs as well as NEO4J and MongoDB. 

These are all databases which support transactions.

Does it make any sense to publish those internal events into e.g. Kafka / Rabbit MQ or if one wants to do this (for whatever reason) one should probably not use Spring Modulith in the first place but go via micro-services oriented architecture because then one has network transport layer in between?

**Comment by odrotbohm** [comment, marked as answer]
The support for this is described in [this section](https://docs.spring.io/spring-modulith/reference/events.html#externalization) of our reference documentation.
